### DISCUSSION TITLE
git question

### DISCUSSION THREAD

**Discussion by sedetweiler** [hlky/stable-diffusion-webui#263] [Q&A]
As I really enjoy this branch, is there a simple way for me to update my local installation from the repo so I can play with the latest features?  I wasn't sure how to use git fetch or git pull properly to have my local version update so it wouldn't trash the models, etc.

**Comment by hlky** [comment]
There's a different directory structure on this dev repo to simplify things and I use a github action to sync things to the right place, check `.github/sync.yml` for what goes where.

Generally it's only webui.py that changes, so you could just pull the changes from the dev repo and copy webui.py.
Yesterday there were 2 different things that absolutely required you to pull from the main repo first though, I did put a notice in the readme and I will again if there's anything like that in the future.

**Comment by sedetweiler** [comment]
So, pretend I have no idea what the proper git command would be, what should I use?  As a side note, I am going to be featuring this git on my YouTube soon, as I think you are ahead of the pack in features and being responsive.  I want to be sure I don't end up being a help desk on updates, so I wanted to ask the painfully obvious questions so I can pass this info on.

**Reply by hlky** [reply]
Honestly I would say just tell people to use the stable repo, changes are never too far behind.

Also if you're going to be promoting it to more people there's even more reason to try and keep as many people as possible, basically everyone who isn't a developer or is helping to test a feature/bug fix, using the stable repo.

https://github.com/hlky/stable-diffusion

If people use this then all they need to do `git pull`

**Reply by hlky** [reply]
If you'd like, when you've finished that part of your video, I will be happy to review it for you before you release it and tell you if anything needs to be changed

**Comment by sedetweiler** [comment]
I don't know how to call this sync file using git.

**Reply by hlky** [reply]
I don't think github actions can be used locally.

**Reply by sedetweiler** [reply]
I have git installed and used clone to get started with this branch.

**Comment by micwalk** [comment]
I've created PR #549 to replicate the sync action locally.
